### ISSUE TITLE
[docker-compose] Only rebuild target "service" on container rebuild

### ISSUE THREAD

**Issue by philipkimmey** [microsoft/vscode-remote-release#2795]
For repositories that have multiple containers, it should be possible to limit rebuilds to the single `service` targeted by `devcontainer.json` when using `docker-compose`.

Given a repository like:

```
monorepo/
  monorepo.code-workspace
  docker-compose.yml
  backend/
    Dockerfile
    .devcontainer/
      devcontainer.json
  frontend/
    Dockerfile
    .devcontainer/ 
      devcontainer.json
```

Modifications to either `devcontainer.json` will require triggering a rebuild, but doing so appears to run basically the raw `docker-compose build` command and then remove/start all containers. (As opposed to `docker-compose build backend` or `docker-compose build frontend`.)

In turn, if one VSCode window is open for each of the containers the container not being rebuilt will lose connection to the container since it's been destroyed.

This problem will be more acute if the improved multi-container support in #1460 is implemented but could also be solved separately.

**Comment by chrmarti** [comment]
If your Dev Container depends on the other containers, you could try using `"runServices": ["frontend"]` in the devcontainer.json.

**Comment by MtDalPizzol** [comment]
Is there any news on this? I just needed to rebuild a single fresh container and I was only able to do that after removing the container and the image via Docker Desktop em reopening the workspace in the devcontainer.

**Comment by chrmarti** [comment]
@MtDalPizzol Not sure you are asking for the same as above. Maybe the 'Rebuild Without Cache' command does what you need? (Since you had to remove the image, 'Rebuild Without Cache' will also pull any updated base image first.)

**Comment by MtDalPizzol** [comment]
@chrmarti Nope... I had 4 containers from 4 compose services on the `.devcontainer` file and only needed to fully rebuild one of them. Since the feature @philipkimmey suggested doesn't exist yet, Rebuild Without Cache rebuilt all 4 containers, including the ones that didn't needed a rebuild.

**Comment by dss010101** [comment]
this is problematic.  is there a work around from command line perhaps?
